### PR TITLE
[DEV-11281] Implement "accordion" UI element for the overlay menu

### PR DESCRIPTION
**Pull request by fgyimah**
https://github.com/prezly/slate/assets/43710420/350fb10c-b377-4c60-a6bc-f1be63cb9bf6

### CODE REVIEW

File: packages/slate-editor/src/components/Accordion/Accordion.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        <div className={styles.Accordion}>
+            <button className={styles.button} onClick={toggleOpen}>
+                <span>{title}</span>
+                {
+                    <Caret
+                        className={classNames(styles.icon, {
+                            [styles.open]: isOpen,
+                        })}
+                    />
+                }
```
Comment (e1himself) [inline packages/slate-editor/src/components/Accordion/Accordion.tsx:30]: Are `{ }` braces needed?

```suggestion
                <Caret
                    className={classNames(styles.icon, {
                        [styles.open]: isOpen,
                    })}
                />
```

File: packages/slate-editor/src/components/Accordion/Accordion.module.scss [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    padding: 0 $spacing-2 $spacing-2;
+
+    &.open {
+        display: block;
+    }
+}
+
+.icon {
+    fill: $white;
+    transition: transform 0.3s ease-in;
```
Comment (e1himself) [inline packages/slate-editor/src/components/Accordion/Accordion.module.scss:32]: Let's speed it up to `0.1s` and use ease-in-out:

```suggestion
    transition: transform 0.1s ease-in-out;
```

**Review by e1himself — APPROVED** [review]
Nice! 

I have a couple of minor comments. 

Otherwise, very good job :+1:

File: packages/slate-editor/src/components/Accordion/Accordion.module.scss (hunk 2 above)
Comment (fgyimah) [reply packages/slate-editor/src/components/Accordion/Accordion.module.scss]: Updated

File: packages/slate-editor/src/components/Accordion/Accordion.tsx (hunk 1 above)
Comment (fgyimah) [reply packages/slate-editor/src/components/Accordion/Accordion.tsx]: Updated

**Pull request merged**